src/Acuminator/Acuminator.Analyzers/Refactorings/EventHandlerSignature/EventHandlerSignatureRefactoring.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.Refactorings.EventHandlerSignature
+{
+	[ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerSignatureRefactoring : CodeRefactoringProvider
+	{
+		private const string ArgsParameterName = "e";
+		private const string EventHandlerMethodName = "_";
+
+		public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+			var pxContext = new PXContext(semanticModel.Compilation);
```

Comment:
@dropsonic no check of semantic model for null or cancellation.
Also no check (at least explicit) for the case when PX.Data is not present in the solution

Reply:
@dropsonic it's better not to rely to much on internal implementation

Reply:
@SENya1990 I still didn't get it - what's wrong with cancellation?
Roslyn documentation doesn't state that `SemanticModel` can be null, and we've never faced such situation on practice.